docs/deployment-examples/custom-scripts/script-modules.md

```diff
@@ -82,14 +67,144 @@ Make sure to select a **Role**, an **Environment** and to put a **Step Name**
 
 ![](script-modules-deployment-release.png)
 
-:::hint
+
+## PowerShell Script Modules{#ScriptModules-PowerShell}
+
+PowerShell script modules get automatically loaded once for every PowerShell script step on your deployment process - the functions and cmdlets will automatically be
+in scope for your script.
+
+Script Module:
+```powershell
+function Say-Hello($name) {
+    Write-output "Hello $name. Welcome to Octopus!"
+}
+```
+
+Script Step:
+```powershell
+Say-Hello -name "George"
+```
+
+### Gotcha
+If you process has 4 steps (i.e Stop IIS, Backup ,Deploy, Start IIS) the entire module will get loaded once at the start of each step. Because of this we encourage users to avoid putting code outside functions on your Script Modules. This way the code will only get executed when the function is properly called from a PowerShell Script step.
+
+In the example Script Module below, the first line which attempts to stop the service "ImportantService" will run for every PowerShell-Script-based step on your deployment . The first time it might succeed (stopping the service), but the subsequent tries will most likely make the overall deployment fail.
```

Comment:
```suggestion
In the example Script Module below, the first line which attempts to stop the service "ImportantService" will run for every PowerShell-Script-based step on your deployment. The first time it might succeed (stopping the service), but the subsequent tries will most likely make the overall deployment fail.
```